packages/fuels/src/cli/utils/fuelsVersionCache.ts

```diff
@@ -0,0 +1,31 @@
+import fs from 'fs';
+import path from 'path';
+
+export const FUELS_VERSION_CACHE_FILE = path.join(__dirname, 'FUELS_VERSION');
+
+export type FuelsVersionCache = string;
+
+export const saveToCache = (cache: FuelsVersionCache) => {
+  fs.writeFileSync(FUELS_VERSION_CACHE_FILE, cache, 'utf-8');
+};
+
+export const FUELS_VERSION_CACHE_TTL = 6 * 60 * 60 * 1000; // 6 hours in milliseconds
+
+export const checkAndLoadCache = (): FuelsVersionCache | null => {
+  const doesVersionCacheExist = fs.existsSync(FUELS_VERSION_CACHE_FILE);
+
+  if (doesVersionCacheExist) {
+    const cachedVersion = fs.readFileSync(FUELS_VERSION_CACHE_FILE, 'utf-8').trim();
+
+    if (!cachedVersion) {
+      return null;
+    }
+
+    const { mtimeMs: cacheTimestamp } = fs.statSync(FUELS_VERSION_CACHE_FILE);
+    const hasCacheExpired = Date.now() - cacheTimestamp > FUELS_VERSION_CACHE_TTL;
+
+    return hasCacheExpired ? null : cachedVersion;
```

Comment:
Is it worth saving the timestamp along the cache so we don't need to call `statSync` ?

Reply:
My initial approach was exactly that but after a round of reviews from the team, I was suggested to move to this approach